Use spf13/cobra for docker login and docker logout

vdemeester opened the pull request:
Moves _registry_ command login and logout to `api/client/registry/login.go` and `api/client/registry/logout.go` and use cobra  🐍.

Also moved some _utils_ method in better files.

/cc @dnephin @thaJeztah @LK4D4 @cpuguy83 

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

dnephin commented on cli/required.go:
```diff
@@ -43,6 +43,23 @@ func RequiresMinArgs(min int) cobra.PositionalArgs {
 }
 
+// RequiresMaxArgs returns an error if there is not at most min args
+func RequiresMaxArgs(min int) cobra.PositionalArgs {
```
The parameter should probably be `max` otherwise it's kind of confusing

vdemeester replied:
oh damn you're absolutely right :sweat: 
/me was in *robot* mode I guess :sweat_smile:

dnephin commented on api/client/login.go:
```diff
@@ -1,184 +0,0 @@
-	// On Windows, force the use of the regular OS stdin stream. Fixes #14336/#14210
-	if runtime.GOOS == "windows" {
-		cli.in = os.Stdin
-	}
```
Is this not necessary anymore?

vdemeester replied:
ah no it is still necessary :stuck_out_tongue_closed_eyes:

vdemeester replied:
Given where it was used, moved it there : https://github.com/docker/docker/pull/23265/files#diff-2b2e46176d739e8721f07f3d736e13f8R86

mlaventure commented:
This already needs a rebase unfortunately @vdemeester

dmcgowan commented:
LGTM

thaJeztah commented:
LGTM

ping @runcom PTAL as well

thaJeztah commented:
also, I think the commits should be squashed, because some changes are in the wrong commit

runcom commented:
LGTM but this needs a rebase :(

vdemeester committed:
Migrate login & logout command to cobra

Also moves some common stuff around :
- `api/client/registry.go` for registry related
method (`ElectAuthServer`, …)
- `api/client/credentials.go` to interact with credentials

Migrate logout command to cobra

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

LK4D4 commented:
LGTM
